File: frontend/src/types/entry.ts
export interface Entry {
  id: number;
  name: string | null;
  type: string;
  volume: number;
  drink_datetime: string;
  bar: string | null;
  rating: number | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface EntryCreate {
  name?: string | null;
  type: string;
  volume: number;
  drink_datetime: string;
  bar?: string | null;
  rating?: number | null;
  notes?: string | null;
}

export interface EntryUpdate {
  name?: string | null;
  type?: string | null;
  volume?: number | null;
  drink_datetime?: string | null;
  bar?: string | null;
  rating?: number | null;
  notes?: string | null;
}

export interface Venue {
  date: string;
  bar: string | null;
  entries: Entry[];
}

export interface StatsSummary {
  weekly_count: number;
  monthly_count: number;
  top_types: string[];
  top_names: string[];
  current_streak_days: number;
}

File: backend/app/models/entry.py
from __future__ import annotations

from datetime import datetime

from sqlalchemy import ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func

from app.db.session import Base


class Entry(Base):
    __tablename__ = "entry"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("user.id"), nullable=False, index=True)
    name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    type: Mapped[str] = mapped_column(String(100), nullable=False)
    volume: Mapped[float] = mapped_column(nullable=False)
    drink_datetime: Mapped[datetime] = mapped_column(nullable=False, default=func.now())
    bar_id: Mapped[int] = mapped_column(ForeignKey("bar.id"), nullable=False, index=True)
    rating: Mapped[float | None] = mapped_column(nullable=True)  # 0.0 – 5.0
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(server_default=func.now(), onupdate=func.now(), nullable=False)
